sql/tests: TestRandomSyntaxGeneration failed

[(sql/tests).TestRandomSyntaxGeneration failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2917225&tab=buildLog) on [release-20.2@72598c05a532dd4617dbd8304c591c69847d4436](https://github.com/cockroachdb/cockroach/commits/72598c05a532dd4617dbd8304c591c69847d4436):

```
    rsg_test.go:728: 1m15s of 5m0s: 313085 executions, 4160 successful
    rsg_test.go:728: 1m20s of 5m0s: 343293 executions, 4563 successful
    rsg_test.go:728: 1m25s of 5m0s: 373352 executions, 4982 successful
    rsg_test.go:728: 1m30s of 5m0s: 387392 executions, 5146 successful
    rsg_test.go:728: 1m35s of 5m0s: 404069 executions, 5348 successful
    rsg_test.go:728: 1m40s of 5m0s: 434287 executions, 5746 successful
    rsg_test.go:728: 1m45s of 5m0s: 463945 executions, 6132 successful
    rsg_test.go:728: 1m50s of 5m0s: 493778 executions, 6524 successful
    rsg_test.go:728: 1m55s of 5m0s: 523235 executions, 6918 successful
    rsg_test.go:728: 2m0s of 5m0s: 553184 executions, 7339 successful
    rsg_test.go:728: 2m5s of 5m0s: 582983 executions, 7703 successful
    rsg_test.go:728: 2m10s of 5m0s: 612735 executions, 8110 successful
    rsg_test.go:728: 2m15s of 5m0s: 642933 executions, 8490 successful
    rsg_test.go:728: 2m20s of 5m0s: 672989 executions, 8861 successful
    rsg_test.go:728: 2m25s of 5m0s: 702690 executions, 9225 successful
    rsg_test.go:728: 2m30s of 5m0s: 732258 executions, 9628 successful
    rsg_test.go:728: 2m35s of 5m0s: 762115 executions, 10015 successful
    rsg_test.go:728: 2m40s of 5m0s: 790888 executions, 10374 successful
    rsg_test.go:728: 2m45s of 5m0s: 820338 executions, 10746 successful
    rsg_test.go:728: 2m50s of 5m0s: 849338 executions, 11111 successful
    rsg_test.go:728: 2m55s of 5m0s: 877694 executions, 11428 successful
    rsg_test.go:728: 3m0s of 5m0s: 906060 executions, 11801 successful
    rsg_test.go:728: 3m5s of 5m0s: 935403 executions, 12172 successful
    rsg_test.go:728: 3m10s of 5m0s: 964232 executions, 12504 successful
    rsg_test.go:728: 3m15s of 5m0s: 993582 executions, 12853 successful
    rsg_test.go:728: 3m20s of 5m0s: 1023677 executions, 13226 successful
    rsg_test.go:728: 3m25s of 5m0s: 1052880 executions, 13590 successful
    rsg_test.go:728: 3m30s of 5m0s: 1081985 executions, 13950 successful
    rsg_test.go:728: 3m35s of 5m0s: 1111077 executions, 14316 successful
    rsg_test.go:728: 3m40s of 5m0s: 1140844 executions, 14694 successful
    rsg_test.go:728: 3m45s of 5m0s: 1144228 executions, 14733 successful
    rsg_test.go:728: 3m50s of 5m0s: 1170511 executions, 15074 successful
    rsg_test.go:728: 3m55s of 5m0s: 1199959 executions, 15428 successful
    rsg_test.go:728: 4m0s of 5m0s: 1229747 executions, 15764 successful
    rsg_test.go:728: 4m5s of 5m0s: 1258889 executions, 16119 successful
    rsg_test.go:728: 4m10s of 5m0s: 1287598 executions, 16458 successful
    rsg_test.go:728: 4m15s of 5m0s: 1317016 executions, 16814 successful
    rsg_test.go:728: 4m20s of 5m0s: 1336942 executions, 17063 successful
    rsg_test.go:728: 4m25s of 5m0s: 1339428 executions, 17101 successful
    rsg_test.go:728: 4m30s of 5m0s: 1347167 executions, 17216 successful
    rsg_test.go:728: 4m35s of 5m0s: 1347976 executions, 17230 successful
    rsg_test.go:728: 4m40s of 5m0s: 1352278 executions, 17288 successful
    rsg_test.go:728: 4m45s of 5m0s: 1382230 executions, 17671 successful
    rsg_test.go:728: 4m50s of 5m0s: 1411554 executions, 18027 successful
    rsg_test.go:728: 4m55s of 5m0s: 1440684 executions, 18407 successful
    rsg_test.go:728: 5m0s of 5m0s: 1466062 executions, 18694 successful
    rsg_test.go:761: 1466070 executions, 18696 successful
    rsg_test.go:769: cannot parse output of Format: sql="PREPARE EXPERIMENTAL_REPLICA AS RESUME JOBS FOR SCHEDULES WITH RECURSIVE ident ( MULTIPOINTZM , SIMPLE , BOOLEAN ) AS NOT MATERIALIZED ( ALTER USER 'string' WITH PASSWORD PLACEHOLDER ) , ident AS ( RESUME JOB error ) TABLE ONLY BOOLEAN . OVERLAY . INTO WITH ORDINALITY", formattedSQL="PREPARE experimental_replica AS RESUME JOBS FOR SCHEDULES WITH RECURSIVE ident (multipointzm, simple, \"boolean\") AS NOT MATERIALIZED (ALTER USER 'string' WITH PASSWORD *****), ident AS (RESUME JOBS VALUES (error)) TABLE \"boolean\".\"overlay\".\"into\" WITH ORDINALITY": at or near "*": syntax error
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRandomSyntaxGeneration087846722
--- FAIL: TestRandomSyntaxGeneration (305.29s)

```

<details><summary>More</summary><p>


```
make stressrace TESTS=TestRandomSyntaxGeneration PKG=./pkg/sql/tests TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #64009 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-21.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-21.1)

- #63331 sql/tests: TestRandomSyntaxGeneration failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestRandomSyntaxGeneration.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/sem/tree/create.go
+++ b/pkg/sql/sem/tree/create.go
@@ -1510,7 +1510,7 @@ func (o *KVOptions) formatAsRoleOptions(ctx *FmtCtx) {
 			if ctx.flags.HasFlags(FmtShowPasswords) {
 				ctx.FormatNode(option.Value)
 			} else {
-				ctx.WriteString("*****")
+				ctx.WriteString("'*****'")
 			}
 		} else if option.Value == DNull {
 			ctx.WriteString(" ")

--- a/pkg/sql/sem/tree/format_test.go
+++ b/pkg/sql/sem/tree/format_test.go
@@ -36,7 +36,7 @@ func TestFormatStatement(t *testing.T) {
 		expected string
 	}{
 		{`CREATE USER foo WITH PASSWORD 'bar'`, tree.FmtSimple,
-			`CREATE USER 'foo' WITH PASSWORD *****`},
+			`CREATE USER 'foo' WITH PASSWORD '*****'`},
 		{`CREATE USER foo WITH PASSWORD 'bar'`, tree.FmtShowPasswords,
 			`CREATE USER 'foo' WITH PASSWORD 'bar'`},
 

